lib/renderer/init.js

```diff
@@ -39,10 +39,16 @@ electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_SYNC_WEB_FRAME_METHOD', (eve
 
 electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', (event, requestId, method, args) => {
   const responseCallback = function (result) {
-    event.sender.send(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, result)
+    Promise.resolve(result)
+      .then((resolvedResult) => {
+        event.sender.send(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, resolvedResult)
+      })
+      .catch((resolvedError) => {
+        console.error(`An async web frame method (${method}) returned a promise that threw an error: `, resolvedError)
```

Comment:
It seems like you could serialize the error, send it over the wire, then return a `Promise.reject(new Error(thatError))`, no?